ipmininet/router/config/ospf6.py

```diff
@@ -0,0 +1,96 @@
+"""Base classes to configure an OSPF6 daemon"""
+
+from ipaddress import ip_interface
+
+from ipmininet.link import address_comparator
+from ipmininet.utils import otherIntf, L3Router, realIntfList
+from .ospf import OSPFRedistributedRoute
+from .utils import ConfigDict
+from .zebra import QuaggaDaemon, Zebra
+
+
+class OSPF6(QuaggaDaemon):
+    """This class provides a simple configuration for an OSPF6 daemon.
+    It advertizes one network per interface (the primary one), and set
+    interfaces not facing another L3Router to passive"""
+    NAME = 'ospf6d'
+    DEPENDS = (Zebra,)
+    last_routerid = ip_interface("0.0.0.1")
+    routerids_taken = []
+
+    def __init__(self, *args, **kwargs):
+        super(OSPF6, self).__init__(*args, **kwargs)
+
+    def build(self):
+        cfg = super(OSPF6, self).build()
+        cfg.redistribute = self.options.redistribute
+        cfg.router_id = self.routerid
+        interfaces = [itf
+                      for itf in realIntfList(self._node)]
+        cfg.interfaces = self._build_interfaces(interfaces)
+        cfg.networks = []
+        return cfg
+
+    def _build_interfaces(self, interfaces):
+        """Return the list of OSPF6 interface properties from the list of
+        active interfaces"""
+        return [ConfigDict(description=i.describe,
+                           name=i.name,
+                           # Is the interface between two routers?
+                           active=self.is_active_interface(i),
+                           priority=i.get('ospf6_priority',
+                                          self.options.priority),
```

Comment:
I think it would be useful to wrap these and try to get OSPFv2 values if specified, i.e.
```python
priority=i.get('ospf6_priority',
                      i.get('ospf_priority', self.options.priority)),
```